drivers/watchdog/wdt_xilinx_axi.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Driver for Xilinx AXI Timebase WDT core, as described in
+ * Xilinx document PG128.
+ *
+ * Note that the window mode of operation of the core is
+ * currently not supported. Also, only a full SoC reset is
+ * supported as a watchdog expiry action.
+ *
+ * Copyright © 2023 Calian Ltd.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT xlnx_xps_timebase_wdt_1_00_a
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/drivers/watchdog/wdt_xilinx_axi.h>
+
+enum xilinx_wdt_axi_register {
+	REG_TWCSR0 = 0x00, /* Control/Status Register 0 */
+	REG_TWCSR1 = 0x04, /* Control/Status Register 1 */
+	REG_TBR = 0x08,	   /* Timebase Register */
+	REG_MWR = 0x0C,	   /* Master Write Control Register */
+};
+
+enum xilinx_wdt_csr0_bits {
+	CSR0_WRS = BIT(3),
+	CSR0_WDS = BIT(2),
+	CSR0_EWDT1 = BIT(1),
+	CSR0_EWDT2 = BIT(0),
+};
+
+enum xilinx_wdt_csr1_bits {
+	CSR1_EWDT2 = BIT(0),
+};
+
+enum {
+	TIMER_WIDTH_MIN = 8,
+};
+
+LOG_MODULE_REGISTER(wdt_xilinx_axi, CONFIG_WDT_LOG_LEVEL);
+
+struct xilinx_wdt_axi_config {
+	mem_addr_t base;
+	uint32_t clock_rate;
+	uint8_t timer_width_max;
+	bool enable_once;
+};
+
+struct xilinx_wdt_axi_data {
+	struct k_spinlock lock;
+	bool last_reset_by_wdt;
+	bool timeout_active;
+	bool wdt_started;
+};
+
+static int wdt_xilinx_axi_setup(const struct device *dev, uint8_t options)
+{
+	const struct xilinx_wdt_axi_config *config = dev->config;
+	struct xilinx_wdt_axi_data *data = dev->data;
+	k_spinlock_key_t key = k_spin_lock(&data->lock);
+	int ret;
+
+	if (!data->timeout_active) {
+		ret = -EINVAL;
+		goto out;
+	}
+	if (data->wdt_started) {
```

Comment:
Please use a blank line after ending braces all over.

```suggestion
	}

	if (data->wdt_started) {
```

Reply:
Changed.